### ISSUE TITLE
conflicts with mutilple units

### ISSUE THREAD

**Issue by duffqiu** [coreos/fleet#1245]
Hi,

I configure a unit with conflicts, but i do that to set muliple units in one line as below:

```
Conflicts=activemq-hub-slave1@*.service activemq-hub-master@%i.service activemq-hub-slave2@%i.service
```

but only the first one wokr, the other doesn't work.

I try to another way with mutiple Conflicts lines, one line with one unit,
it seems works
like below:

```
Conflicts=activemq-hub-master@*.service
Conflicts=activemq-hub-slave1@%i.service
Conflicts=activemq-hub-slave2@%i.service
```

what is the problem? it is a bug or not? or it is the systemd bug?

I check systemd spec, it mentions it can be configures with mutilple units in one line.

[systemd spec](http://www.freedesktop.org/software/systemd/man/systemd.unit.html)

the fleet version is 0.9.2 in coreos 647.2

**Comment by duffqiu** [comment]
one more thing, if i start mutilple units at the same time, fleetctl can't find the proper way to assign the machine.

for example, 

one unit with blow conflicts:

```
Conflicts=activemq-hub-master@*.service
Conflicts=activemq-hub-slave1@%i.service
Conflicts=activemq-hub-slave2@%i.service
```

another unit with below  conflicts:

```
Conflicts=activemq-hub-slave1@*.service
Conflicts=activemq-hub-master@%i.service
Conflicts=activemq-hub-slave2@%i.service
```

after start activemq-hub-master@{1..3}.service,
i try to start activemq-hub-slave1@{1..3}.service

sometimes fleetctl will stuck because  (I only 3 machines)
activemq-hub-master@1 assigned to A
activemq-hub-master@2 assigned to B
activemq-hub-master@3 assigned to C

and 
activemq-hub-slave1@1 assigned to B
activemq-hub-slave1@2 assigned to A

then it is impossible to assign activemq-hub-slave1@3 because activemq-hub-master@3 assigned to C

Maybe we need to add some logic to handle the multiple assignment at the same time to aoivd the conflicts.

**Comment by Dynom** [comment]
I'm not an expert in any way, although I've already spend several days struggling getting things to work (fleet, etcd, CoreOS, etc.). It's far from a smooth ride so far. Also more intelligent (re-)location of units would be a very desirable option.

However, to actually get to your question: I've read somewhere that multiple rules count as "either / or" and a single rule means "any / and". I don't recall it specifically mentioned for the `Conflicts` directive, but following this mantra it seems that it's "by design".

So basically this means "either activemq-hub-master OR activemq-hub-slave1 OR activemq-hub-slave2"
```
Conflicts=activemq-hub-master@*.service
Conflicts=activemq-hub-slave1@%i.service
Conflicts=activemq-hub-slave2@%i.service
```

Not sure if it makes sense in your situation, just something I read somewhere on coreos.com

**Comment by bcwaldon** [comment]
It looks like each `Conflicts` option is parsed as one value, regardless of what follows the equals sign [0]. It does seem like all conflicts should be respected, though, effectively acting as an AND, not an OR [1].

[0] https://github.com/coreos/fleet/blob/master/job/job.go#L169
[1] https://github.com/coreos/fleet/blob/master/agent/state.go#L43-L67
